### PR TITLE
Add the review's reviewer to each element of the Request History

### LINKED ISSUE #7669: Show added reviewer name
**Is your feature request related to a problem? Please describe.**
When adding a new review, the name of the reviewer is not shown in the history. It would make sense to see who was added.

**Describe the solution you'd like**
The history element tells you who was added.

**Describe alternatives you've considered**
n/a

**Additional context**
It looks like that at the moment:

review

### PR DESCRIPTION
**Pull request by danidoni**
The name of the reviewer was not being shown and it makes sense to show that
info on the Request's history

Closes #7669

### CODE REVIEW

**Comment by danidoni** [comment]
Do not merge this yet, I'm checking if some flickering tests I have locally happen also on the CircleCI

**Comment by coolo** [comment]
please avoid this N+1 query

File: src/api/app/models/history_element.rb [hunk 1]
```diff
@@ -19,14 +19,18 @@ def color
...
-                    .joins('JOIN history_elements ON history_elements.description_extension = reviews.id') },
              class_name: '::Review',
-             foreign_key: :description_extension
+             foreign_key: :description_extension,
+             scope: :restrict_to_review_added
 
   self.description = 'Request was updated'
   self.abstract_class = true
 
+  def self.restrict_to_review_added
```
Comment (hennevogel) [inline src/api/app/models/history_element.rb]: Sounds like a [scope](https://guides.rubyonrails.org/active_record_querying.html#scopes)

Comment (danidoni) [reply src/api/app/models/history_element.rb]: Oh sorry, I abandoned the idea of extracting that lambda.

**Comment by danidoni** [comment]
> please avoid this N+1 query

@coolo N+1 avoided. Check it out, please.

File: src/api/app/models/bs_request.rb [hunk 2]
```diff
@@ -92,7 +92,9 @@ class SaveError < APIError
   has_many :bs_request_actions, dependent: :destroy
   has_many :reviews, dependent: :delete_all
   has_many :comments, as: :commentable, dependent: :delete_all
-  has_many :request_history_elements, -> { order(:created_at) }, class_name: 'HistoryElement::Request', foreign_key: :op_object_id
+  has_many :request_history_elements, -> { order(:created_at) },
```
Comment (dmarcoux) [inline src/api/app/models/bs_request.rb]: Is this change needed? It seems to be the same code, just split up on multiple lines.

File: src/api/app/views/webui/request/_request_history_element.html.haml [hunk 3]
```diff
@@ -8,6 +8,9 @@
         = link_to(element.user_action, request_show_path(element.description_extension))
       - else
         = element.user_action
+      - if element.respond_to?(:review) && element.review
+        for
+        = render partial: 'webui/request/reviewer', locals: { review: element.review }
```
Comment (hennevogel) [inline src/api/app/views/webui/request/_request_history_element.html.haml]: I think you can work on the design of this a bit

Screenshot from 2019-09-19 17-00-36

The click targets are very close together, the reviewer avatar looks out of place, the sentence always wraps etc.

File: src/api/app/models/bs_request.rb (hunk 2 above)
Comment (danidoni) [reply src/api/app/models/bs_request.rb]: No, it's not really needed. Reverting

File: src/api/app/views/webui/request/_request_history_element.html.haml (hunk 3 above)
Comment (danidoni) [reply src/api/app/views/webui/request/_request_history_element.html.haml]: On it

Comment (danidoni) [reply src/api/app/views/webui/request/_request_history_element.html.haml]: image
What about this?

Comment (coolo) [reply src/api/app/views/webui/request/_request_history_element.html.haml]: that's neatly faked user names. Can you try https://build.opensuse.org/request/show/732250's "factory-auto added review for opensuse-review-team 2 days ago" and "staging-bot added review for openSUSE:Factory:Staging:adi:74 2 days ago" in the little place you got there?

Comment (danidoni) [reply src/api/app/views/webui/request/_request_history_element.html.haml]: Yeah, will try. Thanks!

**Comment by danidoni** [comment]
> > please avoid this N+1 query
> 
> @coolo N+1 avoided. Check it out, please.

Well, I've found out that I really didn't avoid anything, so the N+1 is still there. :angry:

**Comment by hennevogel** [comment]
> Well, I've found out that I really didn't avoid anything, so the N+1 is still there. angry

Because?

**Comment by coolo** [comment]
> Because?

Because HistoryElement doesn't have a `has_many :reviews` - but `HistoryElement::Review` has a review function that is a built n+1. And if you fetch things from the review, it gets full (N+1)*N

**Comment by danidoni** [comment]
I'm closing this PR. See the [referenced issue](https://github.com/openSUSE/open-build-service/issues/7669) for details.
